AGENTS.md

```diff
@@ -0,0 +1,433 @@
+```
+
+**Meson**:
+
+```bash
+meson setup build --cross-file="${MESON_TARGET_TOOLCHAIN}"
+meson compile -C build -j${nproc}
+meson install -C build
+```
+
+**Make**:
+
+```bash
+make -j${nproc} PREFIX=${prefix}
+make install PREFIX=${prefix}
+```
+
+**Go** (add `compilers=[:c, :go]`):
+
+```bash
+go build -o ${bindir}/executable
+```
+
+**Rust** (add `compilers=[:c, :rust]`):
+
+```bash
+cargo build --release
+install -Dm755 target/*/release/executable ${bindir}/executable
+```
+
+### Platform-Specific Logic
+
+```bash
+if [[ "${target}" == *-mingw* ]]; then
+    # Windows-specific
+elif [[ "${target}" == *-apple-* ]]; then
+    # macOS-specific
+elif [[ "${target}" == *-freebsd* ]]; then
+    # FreeBSD-specific
+fi
+
+if [[ ${nbits} == 32 ]]; then
+    # 32-bit specific
+fi
+```
+
+### Products
+
+- **LibraryProduct**: Shared libraries (`.so`, `.dylib`, `.dll`)
+- **ExecutableProduct**: Binary executables
+- **FileProduct**: Other files (headers, data files)
+- **FrameworkProduct**: macOS frameworks
+
+### Dependencies
+
+- **Dependency**: Runtime dependency (will be in JLL package)
+- **BuildDependency**: Build-time only (not in final JLL)
+- **HostBuildDependency**: Runs on build host, not target
+
+Always add `_jll` suffix: `Dependency("Zlib_jll")`
+
+### GCC Version Selection
+
+Use `preferred_gcc_version=v"X"` for:
+
+- **C++ code**: Use oldest GCC that compiles (≤10 for Julia v1.6 compatibility)
+- **Dependencies built with newer GCC**: Match or exceed their GCC version
+- **Musl bugs**: Use GCC ≥6 to avoid `posix_memalign` issues
+- Default is GCC 4.8.5 for maximum compatibility
+
+
+### Optional Arguments
+
+```julia
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+    julia_compat="1.10",                   # Minimum Julia version
+    preferred_gcc_version=v"8",            # GCC version
+    preferred_llvm_version=v"13",          # LLVM version
+    compilers=[:c, :rust],                 # Additional compilers
+    clang_use_lld=false,                   # Use LLD linker
+)
+```
+
+## Common Patterns
+
+### Applying Patches
+
+```julia
+sources = [
+    ArchiveSource("..."),
+    DirectorySource("./bundled"),
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/package-*
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fix.patch
+# ... rest of build
+"""
+```
+
+### Platform Filtering
+
+```julia
+# Only 64-bit platforms
+platforms = filter(p -> arch(p) != "i686" && arch(p) != "armv7l", supported_platforms())
+
+# Expand C++ string ABI variants
+platforms = expand_cxxstring_abis(platforms)
+
+# Specific platforms
+platforms = [Platform("x86_64", "linux"), Platform("x86_64", "macos")]
+```
+
+### Setting Environment Variables
+
+```bash
+export CPPFLAGS="-I${includedir}"
+export LDFLAGS="-L${libdir}"
+export PKG_CONFIG_PATH="${prefix}/lib/pkgconfig:${PKG_CONFIG_PATH}"
+```
+
+### Multiple Source Directories
+
+```bash
+cd ${WORKSPACE}/srcdir
+mkdir build && cd build
+cmake ../package-source
+```
```

Comment:
I am confused what this means by "multiple source directories" - this just looks like a separate build directory for Cmake.

Reply:
Changed to `Out-of-Source Builds`. Is that clearer?